packages/ra-core/src/auth/useGetIdentity.ts

```diff
@@ -45,21 +44,20 @@ const useGetIdentity = () => {
     const authProvider = useAuthProvider();
     useEffect(() => {
         if (authProvider && typeof authProvider.getIdentity === 'function') {
-            const callAuthProvider = async () => {
-                try {
-                    const identity = await authProvider.getIdentity();
```

Comment:
This complained about authProvider.getIdentity possibly being undefined. There is a check, but it's in outer scope, and this is async function which can be executed later, so this check doesn't apply to the inner function. Therefore, I changed it to call authProvider.getIdentity() immediately in the current scope and simply use .then() / .catch() to store the results.